src/embit/descriptor/arguments.py

```diff
@@ -14,7 +14,10 @@ def __init__(self, fingerprint: bytes, derivation: list):
     @classmethod
     def from_string(cls, s: str):
         arr = s.split("/")
-        mfp = unhexlify(arr[0])
+        try:
+            mfp = unhexlify(arr[0])
+        except Exception as err:
+            raise ArgumentError(err) from err
```

Comment:
Just to be sure, did you test in device if `raise from` is functional in micropython?
If it was not used so for by Krux or Embit, it would be good to test.

Reply:
Mmm, good point. Didn't tested. My bad for this lack of attention. I believe `micropython` do not supoort it right? Maybe something like the below can be better?

```suggestion
            raise ArgumentError(str(err))
```

Reply:
Maybe it works the way it is, but it needs to be tested

Reply:
now that you said, maybe these are theoretical cases, but I believe that in real devices, they may not happen. Maybe it's a dead-code?

Reply:
tested in a `wondermv`, but did not get any of these theoretical errors (IMO because the devices, in a real situations, will never load incorrect fingerprints).